### ISSUE TITLE
install bundle errors with latest liberty GM.  

### ISSUE THREAD

**Issue by kevin-ortega** [OpenLiberty/open-liberty#7300]
[messages.log](https://github.com/OpenLiberty/open-liberty/files/3104079/messages.log)
Seeing these errors with latest Liberty GM in liberty-for-java buildpack.   These errors do not occur with 19.0.0.3.   

[4/22/19 16:02:04:332 UTC] 00000043 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.lang.NullPointerException com.ibm.ws.container.service.state.internal.ApplicationStateManager 46" at ffdc_19.04.22_16.02.03.0.log
[4/22/19 16:02:04:438 UTC] 00000039 com.ibm.ws.cdi.config.liberty.CDI12ContainerConfig           W implicit.bean.scanning.disabled.CWOWB1009W
[4/22/19 16:02:04:473 UTC] 00000043 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.lang.NullPointerException com.ibm.ws.container.service.state.internal.ApplicationStateManager 59" at ffdc_19.04.22_16.02.04.0.log
[4/22/19 16:02:04:481 UTC] 00000043 com.ibm.ws.http.internal.VirtualHostImpl                     A CWWKT0017I: Web application removed (default_host): http://localhost:8081/IBMJMXConnectorREST/
[4/22/19 16:02:04:571 UTC] 00000042 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "com.ibm.ws.container.service.state.StateChangeException: java.lang.NullPointerException com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase 565" at ffdc_19.04.22_16.02.03.1.log
.
. 
2019-04-22T12:02:05.96-0400 [APP/PROC/WEB/0] OUT [INFO    ] SRVE0242I: [com.ibm.ws.jmx.connector.server.rest] [/IBMJMXConnectorREST] [JMXRESTProxyServlet]: Initialization successful.
   2019-04-22T12:02:05.99-0400 [APP/PROC/WEB/0] OUT [INFO    ] SRVE0253I: [com.ibm.ws.jmx.connector.server.rest] [/IBMJMXConnectorREST] [JMXRESTProxyServlet]: Destroy successful.
   2019-04-22T12:02:06.11-0400 [APP/PROC/WEB/0] ERR [ERROR   ] CWWKZ0202E: Unable to install bundle com.ibm.ws.rest.handler_1.0.27.cl190420190419-0642 [177] with context root /ibm/api into the web container.
   2019-04-22T12:02:06.11-0400 [APP/PROC/WEB/0] ERR [ERROR   ] CWWKZ0202E: Unable to install bundle com.ibm.ws.jmx.connector.server.rest_1.1.27.cl190420190419-0642 [182] with context root /IBMJMXConnectorREST into the web container.
[ffdc_19.04.22_16.02.03.0.log](https://github.com/OpenLiberty/open-liberty/files/3104069/ffdc_19.04.22_16.02.03.0.log)

**Comment by aguibert** [comment]
It appears that PR https://github.com/OpenLiberty/open-liberty/pull/7017 caused behavior changes to the `ApplicationStateListener` SPI. The methods on this SPI that accept an `ApplicationInfo` as a parameter, such as `public void applicationStarted(ApplicationInfo appInfo)` may be impacted.

### Old behavior:
```java
    @Override
    public void applicationStarted(ApplicationInfo appInfo) throws StateChangeException {
        Object appInfoName = appInfo.getConfigHelper().get("name"); // works OK
        // ...
    }
```

### New behavior:
```java
    @Override
    public void applicationStarted(ApplicationInfo appInfo) throws StateChangeException {
        Object appInfoName = appInfo.getConfigHelper().get("name"); // NPE because getConfigHelper() returns null
        // ...
    }
```

**Comment by marikaj123** [comment]
Issue was found in 19004 and would be fixed at the next release with an ifix in 19004.

**Comment by marikaj123** [comment]
@garypicher @aguibert @gmarcy - What someone provide us with the status for the fix?

**Comment by gmarcy** [comment]
Kevin already confirmed that the TestFix build result was good.  The final iFix build is at https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_k4WzMHGrEemMj7NCS7eaxg&tab=com.ibm.team.build.web.ui.internal.editors.result.overview.OverviewPage

**Comment by marikaj123** [comment]
@gmarcy would Liberty 19005 contain the fix, as well?

**Comment by marikaj123** [comment]
@gmarcy - Let us know what is the PR for 19005.

**Comment by marikaj123** [comment]
@gmarcy - Did the PR make into the Release build.  Please provided here for our reference.

**Comment by kevin-ortega** [comment]
I patched a 19.0.0.5 driver with a jar I obtained from Glenn and I the errors I previously saw are not occurring.

**Comment by kevin-ortega** [comment]
Driver cl190520190517-0758 has the fix for this issue.   Closing.
